refactored the landing page and TOC (#261)
ROCm/rocAL: docs/conceptual/architecture.rst

.. _architecture:

********************************************************************

> rocAL Architecture components

Reviewer: Documentation changes are handled by Sandra. Please work with her to update it and remove it from this PR

Author: Does this have to be removed?

.. _architecture:

********************************************************************

> Architecture components

Reviewer 2: ```suggestion
rocAL Architecture components
```

********************************************************************

> The rocAL architecture comprises rocAL Master-Graph and ROCm Performance Primitive (RPP) as major components.

Reviewer 2: ```suggestion
rocAL is comprised of rocAL Master-Graph and ROCm Performance Primitive (RPP).
```

Reviewer 2: not sure this is better - i have problems with comprise - try to avoid.

Reviewer 2: hmmm - turns out comprise is noted in the google style:
![image](https://github.com/user-attachments/assets/0494d2a4-6b51-47c5-a921-9fe9fd0e6ab5)
